dask/tests/test_base.py

```diff
@@ -255,6 +255,16 @@ def test_tokenize_object():
             normalize_token(o)
 
 
+def test_tokenize_function_cloudpickle():
+    a, b = (lambda x: x, lambda x: x)
+    # No error by default
+    tokenize(a)
+
+    with dask.config.set({"tokenize.ensure-deterministic": True}):
+        with pytest.raises(RuntimeError, match="may not be deterministically hashed"):
+            tokenize(b)
```

Comment:
@ian-r-rose If we check for `tokenize(a)` here, the test fails because the error isn't raised. I'm guessing it's because the `tokenize(a)` call on line 261 gets cached... is that alright?